lib/utils/compose_ts.ts

```diff
@@ -148,11 +150,53 @@ export async function loadProject(
 			);
 			composeStr = compose.defaultComposition(undefined, opts.dockerfilePath);
 		}
+
+		// If local push, merge dev compose overlay
+		if (opts.isLocal) {
+			composeStr = await mergeDevComposeOverlay(
+				logger,
+				composeStr,
+				opts.projectPath,
+			);
+		}
 	}
 	logger.logDebug('Creating project...');
 	return createProject(opts.projectPath, composeStr, opts.projectName);
 }
 
+/**
+ * Check for existence of docker-compose dev overlay file
+ * and merge in services definitions.
+ */
+async function mergeDevComposeOverlay(
+	logger: Logger,
+	composeStr: string,
+	projectRoot: string,
+) {
+	const devOverlayFilename = 'docker-compose.dev.yml';
+	const devOverlayPath = path.join(projectRoot, devOverlayFilename);
+
+	if (await exists(devOverlayPath)) {
+		logger.logInfo(
+			`Docker compose dev overlay detected (${devOverlayFilename}) - merging.`,
+		);
+		try {
+			const compose = yaml.load(composeStr);
+			const devOverlay = yaml.load(await fs.readFile(devOverlayPath, 'utf8'));
+			// We only want to merge the services section
+			compose.services = { ...compose.services, ...devOverlay.services };
```

Comment:
Right, my understanding is that we want a shallow merge.

Reply:
Jumping in on this a bit late, I just noticed the change by chance:
Using `docker-compose`, you can specify multiple yml files using the `-f` flag. It uses a recursive merge as described here: https://docs.docker.com/compose/extends/#adding-and-overriding-configuration

> For single-value options [...], the new value replaces the old value. [...]
> For the multi-value options [...], Compose concatenates both sets of values

I think this would be a preferable merge method, for parity. This way you can have a minimal `docker-compose.dev.yml`, for example, exposing a node debugger in local mode only:
```yml
services:
  app:
    ports:
      - "9229:9229"
```

Reply:
Thanks for the suggestion @edorgeville 👍  This PR has been merged and released (CLI v12.40.0), but issue #1142 is still open and will make further code changes under that issue. May ask you to copy and paste your comment above to that issue? This ensures that we will take it into account.

Reply:
Done, thank you